src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -156,7 +156,7 @@ internal BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, Dia
                 conversion,
                 @checked: false,
                 explicitCastInCode: false,
-                isExplicitlyNullable: false,
+                conversionGroup: null,
```

Comment:
I think you already have a PROTOTYPE comment for deconstructions elsewhere. We'll need to follow-up.

Every scenario that you tested for assignment (`x = expr;`) could also be tested with deconstruction-assignment from tuple (`(x, y) = (expr, expr)`).
Also there are implicit conversion scenarios (that could be user-defined) when using a tuple type or `Deconstruct` method. For instance, `(i, j) = expr` and you have `expr.Deconstruct(out C, out C)` and a user-defined conversion from `C` to `int`.

Then there are deconstruction-foreach (`foreach ((int i, int j) in collection) ...`) which could also report nullability warnings.